receiver/mysqlreceiver/config_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysqlreceiver
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/multierr"
+)
+
+func TestValidate(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		cfg      *Config
+		expected error
+	}{
+		{
+			desc: "missing username and password",
+			cfg:  &Config{},
+			expected: multierr.Combine(
+				errors.New(ErrNoUsername),
+				errors.New(ErrNoPassword),
+			),
```

Comment:
easier to use the text that you expect and use assert.EqualError?

Reply:
In my opinion this pattern is nice. It allows for "no error" scenarios by just setting `expected: nil`.